core/types/src/error.rs

```diff
@@ -231,147 +235,165 @@ pub enum Error {
-    unsupported_att_key_id = { sgx_status_t::SGX_ERROR_UNSUPPORTED_ATT_KEY_ID, Error::UnsupportedAttKeyId },
-    platform_cert_unavailable = { sgx_status_t::SGX_ERROR_PLATFORM_CERT_UNAVAILABLE, Error::PlatformCertUnavailable },
-    interrupted = { sgx_status_t::SGX_INTERNAL_ERROR_ENCLAVE_CREATE_INTERRUPTED, Error::EnclaveCreateInterrupted },
+        unexpected = { sgx_status_t::SGX_ERROR_UNEXPECTED, Error::Unexpected },
+        memory_map = { sgx_status_t::SGX_ERROR_MEMORY_MAP_FAILURE, Error::MemoryMapFailure },
+        invalid_function = { sgx_status_t::SGX_ERROR_INVALID_FUNCTION, Error::InvalidFunction },
+        stack_overrun = { sgx_status_t::SGX_ERROR_STACK_OVERRUN, Error::StackOverrun },
+        undefined_symbol = { sgx_status_t::SGX_ERROR_UNDEFINED_SYMBOL, Error::UndefinedSymbol },
+        invalid_launch_token = { sgx_status_t::SGX_ERROR_INVALID_LAUNCH_TOKEN, Error::InvalidLaunchToken },
+        mac_mismatch = { sgx_status_t::SGX_ERROR_MAC_MISMATCH, Error::MacMismatch },
+        invalid_keyname = { sgx_status_t::SGX_ERROR_INVALID_KEYNAME, Error::InvalidKeyname },
+        service_unavailable = { sgx_status_t::SGX_ERROR_SERVICE_UNAVAILABLE, Error::ServiceUnavailable },
+        unsupported_config = { sgx_status_t::SGX_ERROR_UNSUPPORTED_CONFIG, Error::UnsupportedConfig },
+        no_privilege = { sgx_status_t::SGX_ERROR_NO_PRIVILEGE, Error::NoPrivilege },
+        pcl_encrypted = { sgx_status_t::SGX_ERROR_PCL_ENCRYPTED, Error::PclEncrypted },
+        pcl_guid_mismatch = { sgx_status_t::SGX_ERROR_PCL_GUID_MISMATCH, Error::PclGuidMismatch },
+        file_bad_status = { sgx_status_t::SGX_ERROR_FILE_BAD_STATUS, Error::FileBadStatus },
+        file_close_failed = { sgx_status_t::SGX_ERROR_FILE_CLOSE_FAILED, Error::FileCloseFailed },
+        unsupported_att_key_id = { sgx_status_t::SGX_ERROR_UNSUPPORTED_ATT_KEY_ID, Error::UnsupportedAttKeyId },
+        platform_cert_unavailable = { sgx_status_t::SGX_ERROR_PLATFORM_CERT_UNAVAILABLE, Error::PlatformCertUnavailable },
+        interrupted = { sgx_status_t::SGX_INTERNAL_ERROR_ENCLAVE_CREATE_INTERRUPTED, Error::EnclaveCreateInterrupted },
     )]
     fn from_sgx_to_error(sgx_status: sgx_status_t, expected: Error) {
-        let error: Error = sgx_status.into();
-        assert_eq!(error, expected);
+        assert_eq!(
+            Error::try_from(sgx_status).expect("Could not convert SGX Status to an error"),
+            expected
+        );
     }
 
     #[test]
     fn unknown_sgx_error_maps_to_unexpected() {
         let unknown = sgx_status_t(0x8000);
-        let error: Error = unknown.into();
-        assert_eq!(error, Error::Unexpected);
+        assert_eq!(
+            Error::try_from(unknown).expect("Could not parse an unknown SGX Status"),
+            Error::Unexpected
+        );
+    }
+
+    #[test]
+    fn success_is_not_an_error() {
+        assert!(Error::try_from(sgx_status_t::SGX_SUCCESS).is_err())
```

Comment:
:wrench: With the name of this test would it better to be something like
```suggestion
    fn success_is_not_an_error() {
        assert!(sgx_status_t::SGX_SUCCESS.into_result().is_ok())
```